sqs/src/main/scala/akka/stream/alpakka/sqs/SqsBatchAckFlowStage.scala

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.sqs
+
+import akka.stream.alpakka.sqs.SqsBatchAckFlowStage._
+import akka.stream.alpakka.sqs.scaladsl.AckResult
+import akka.stream.stage._
+import akka.stream.{Attributes, FlowShape, Inlet, Outlet}
+import com.amazonaws.handlers.AsyncHandler
+import com.amazonaws.services.sqs.AmazonSQSAsync
+import com.amazonaws.services.sqs.model._
+
+import scala.collection.JavaConverters._
+import scala.concurrent.{Future, Promise}
+import scala.util.{Failure, Success, Try}
+
+private[sqs] final class SqsBatchAckFlowStage[A](queueUrl: String, sqsClient: AmazonSQSAsync)(
```

Comment:
Does SqsBatchAckFlowStage need to be generic on A? 
Couldn't it just require the Pair type and get away without the implicit extractor functions?

Reply:
The stage is a `Flow[(Call, Seq[A]), _]`. If it were a `Flow[Seq[Pair], _]` the seq could contain incompatible messages in the same batch.

That's the reason, but I agree whith you on that it can be improved.

I'll try to push a simpler approach later today.